// jwks-rsa/src/index.js
const { JwksClient } = require('jwks-rsa/src/JwksClient');
const errors = require('jwks-rsa/src/errors');
const { hapiJwt2Key, hapiJwt2KeyAsync } = require('jwks-rsa/src/integrations/hapi');
const { expressJwtSecret } = require('jwks-rsa/src/integrations/express');
const { koaJwtSecret } = require('jwks-rsa/src/integrations/koa');
const { passportJwtSecret } = require('jwks-rsa/src/integrations/passport');

module.exports = (options) => {
  return new JwksClient(options);
};
module.exports.JwksClient = JwksClient;

module.exports.ArgumentError = errors.ArgumentError;
module.exports.JwksError = errors.JwksError;
module.exports.JwksRateLimitError = errors.JwksRateLimitError;
module.exports.SigningKeyNotFoundError = errors.SigningKeyNotFoundError;

module.exports.expressJwtSecret = expressJwtSecret;
module.exports.hapiJwt2Key = hapiJwt2Key;
module.exports.hapiJwt2KeyAsync = hapiJwt2KeyAsync;
module.exports.koaJwtSecret = koaJwtSecret;
module.exports.passportJwtSecret = passportJwtSecret;
